packages/server/src/fhir/operations/codesystemvalidatecode.ts

```diff
@@ -62,19 +62,38 @@ export async function codeSystemValidateCodeHandler(req: FhirRequest): Promise<F
 }
 
 export async function validateCoding(codeSystem: CodeSystem, coding: Coding): Promise<Coding | undefined> {
-  if (coding.system && coding.system !== codeSystem.url) {
-    return undefined;
+  return validateCodings(codeSystem, [coding]).then((results) => results[0]);
+}
+
+export async function validateCodings(codeSystem: CodeSystem, codings: Coding[]): Promise<(Coding | undefined)[]> {
+  const eligible: boolean[] = new Array(codings.length);
+  const codesToQuery = new Set<string>();
+  for (let i = 0; i < codings.length; i++) {
+    const c = codings[i];
+    if (c.system && c.system !== codeSystem.url) {
+      continue;
+    }
+    if (c.code) {
+      codesToQuery.add(c.code);
+      eligible[i] = true;
+    }
   }
 
-  const query = new SelectQuery('Coding')
-    .column('id')
-    .column('display')
-    .where('code', '=', coding.code)
-    .where('system', '=', codeSystem.id);
+  let result: any[] | undefined;
+  if (codesToQuery.size > 0) {
+    const query = new SelectQuery('Coding')
+      .column('id')
+      .column('code')
+      .column('display')
+      .where('code', 'IN', codesToQuery)
+      .where('system', '=', codeSystem.id);
+
+    const db = getDatabasePool();
+    result = await query.execute(db);
+  }
 
-  const db = getDatabasePool();
-  const result = await query.execute(db);
-  return result.length
-    ? { id: result[0].id, system: codeSystem.url, code: coding.code, display: result[0].display }
-    : undefined;
+  return codings.map((c, idx) => {
+    const row = eligible[idx] && result?.find((r: any) => r.code === c.code);
```

Comment:
I believe if you use `Coding` on line 82, you could remove the type from `(r)` entirely.